java/google/registry/whois/DomainLookupCommand.java

```diff
@@ -31,7 +34,8 @@ public DomainLookupCommand(InternetDomainName domainName, @Nullable InternetDoma
   }
 
   @Override
-  WhoisResponse getSuccessResponse(DomainResource domain, DateTime now) {
-    return new DomainWhoisResponse(domain, now);
+  protected Optional<WhoisResponse> getSuccessResponse(InternetDomainName domainName, DateTime now) {
+    final DomainResource domainResource = loadByForeignKey(DomainResource.class, domainName.toString(), now);
+    return domainResource == null ? Optional.absent() : Optional.of(new DomainWhoisResponse(domainResource, now));
```

Comment:
Yeah, it's not as obvious of an improvement as if the ternary weren't required.  You can still do this:

```
return Optional.fromNullable(domainResource == null ? null : new DomainWhoisResponse(domainResource, now));
```

But it's not a big improvement.